### DISCUSSION TITLE
Accept both subdir and full path in config for templates, dailies, weeklies, images

### DISCUSSION THREAD

**Discussion by lambtho12** [nvim-telekasten/telekasten.nvim#70] [Ideas]
As I briefly suggested in https://github.com/renerocksai/telekasten.nvim/issues/69, I think it might be a good idea to support both subdirectories or full path in the configuration for the various notes directories (templates, dailies, weeklies and images). As you did not address that suggestion when solving the issue, I think we might discuss this here.

This feature would not add too much complexity: simply check if the `M.Cfg.<option>` is an absolute path or not; if not, convert relative paths to absolute and always use absolute ones everywhere.

This would allow anyone to pick their preferred way (all directories within the Zettlekasten home, or directory spread everywhere for various reasons). With both options being supported, this would also **keep compatibility** with the current options.

Maybe I have not put enough thought into this, but I do not really see any reason not to do that.

**Comment by renerocksai** [comment]
While I don't really see a pressing need for that since the config example shows how to make absolute paths, doing it is only a matter of time investment.

It would also be kind of more logical because currently, the image subdir is an exception to all the others. So if we make all dir settings optional subdirs, care needs to be taken everywhere the image subdir is used, as there a real subdir is expected. But that shouldn't be too hard.

**Reply by lambtho12** [reply]
Indeed, this is not a pressing matter. 

> care needs to be taken everywhere the image subdir is used, as there a real subdir is expected

I already had a quick look, actually the only times where the `image_subdir` is used, it is to convert it to a full path before using it 😄 . See below:

https://github.com/renerocksai/telekasten.nvim/blob/b8767dd9e3e28c357ae90c7fffc5d5653c4d8931/lua/telekasten.lua#L192-L195

https://github.com/renerocksai/telekasten.nvim/blob/b8767dd9e3e28c357ae90c7fffc5d5653c4d8931/lua/telekasten.lua#L253-L256


I will make a PR for that whenever I am done with the renaming notes feature.

**Reply by renerocksai** [reply]
I really appreciate your PRs! This is awesome!

**Reply by lambtho12** [reply]
Glad I can contribute. This is the first time I actually do some meaningful PRs on a project that is not mine.

As an engineer working in academia, my codes are usually poorly-written garbage to compute stuff (they still work though).

**Reply by renerocksai** [reply]
I can relate to that: as a SW engineer working at a research institute ... I see throwaway code or code the authors don't understand themselves every day. But I see it this way: this is where engineers (of all disciplines) can make a huge difference. I try my best to make super awesome solutions that enable super awesome research projects that wouldn't be possible otherwise because of a lack of programming skills (and experience, ...) - because most researchers I know aren't good at writing real-world software - and bringing in third parties is often not feasible due to time- and budget constraints. Mind you, telekasten is a mere hobby project so it doesn't get the time it probably would deserve...

And that's why your contributions are even more important and welcome! Yayyyy :smile:

**Comment by renerocksai** [comment]
(btw I must have missed your suggestion, sorry 'bout that)
